### DISCUSSION TITLE
Image on docker hub?

### DISCUSSION THREAD

**Discussion by zalmael** [hsakoh/switchbot-mqtt#26] [General]
Hi,

I'm really interested with this project and I would like to know if it's planned to publish switchbot-mqtt on dockerhub ?
Indeed, I have a device on which I can run docker images, but only those available from dockerhub.

I'm not sure what are the prerequisites, and maybe you have your own reasons not to do it, but I had to ask :)

Thank you very much.

Pierre.

**Comment by hsakoh** [comment]
There isn't a strong reason, but...

The add-on I referred to was using the GitHub Container Registry, so I published it similarly.
I'm not very familiar with the Docker ecosystem.

I understand now that there are people who cannot use the GitHub Container Registry and can only obtain images from DockerHub.

I have created a DockerHub account and pushed the image there.
I will try to keep it updated with the latest versions.

**Reply by hsakoh** [reply]
https://hub.docker.com/r/hsakoh/switchbotmqtt-armv7
https://hub.docker.com/r/hsakoh/switchbotmqtt-amd64
https://hub.docker.com/r/hsakoh/switchbotmqtt-aarch64

**Reply by zalmael** [reply]
Wow, that was quick !!! You rock.
Well, unfortunately, it seems that my system (iHost from Sonoff) cannot install it 😄. I will have a look at the logs to see what happens.
Anyway, thank you for the effort and the work you have done !

**Reply by zalmael** [reply]
Hi @hsakoh,
I think I have found the issue. I was looking at the logs my device provides, and I got a 404 error.
```
pull hsakoh/switchbotmqtt-armv7 error 404 {
  'api-version': '1.41',
  'content-type': 'application/json',
  'docker-experimental': 'false',
  ostype: 'linux',
  server: 'Docker/20.10.18 (linux)',
  date: 'Fri, 28 Jun 2024 08:26:48 GMT',
  'content-length': '107',
  connection: 'close'
```
As I  told you, I don't have the entire control on the device and commands, so I can't define which version I want to pull. By default, if we don't specify the version (like in the logs above), pull will check for "latest". But this tag doesn't exist. Is it possible you had a tag named latest ?

Thank you for your help.

Pierre.

**Reply by hsakoh** [reply]
I just now added the latest tag.

**Reply by zalmael** [reply]
Perfect, I can pull it now !!!

**Reply by zalmael** [reply]
Ok, so now I've managed to pull the image, I'm not sure how to run it, I'm really struggling on the different steps (building the app first)  and I'm not sure I will be able to run it on my system... I hope it will at least help others to have made your work available on docker hub :)

**Reply by hsakoh** [reply]
Since I'm not entirely sure about your environment, here are a few general tips:

1. The published add-on's Docker image has the app compiled and the image built.
This means that as soon as you pull it, the steps No.1 and No.2 under "Running on a Separate Docker Container" on the following page are already done:
https://github.com/hsakoh/switchbot-mqtt/blob/main/INSTALLATION.md#install-add-on

2. Next steps:
- Assign a volume to /data inside the container.
- Open port 8099.
   - You can refer to this [docker-compose.yaml](https://github.com/hsakoh/switchbot-mqtt/blob/main/_docker-compose/docker-compose.yaml).
- Place the options.json file in the assigned volume.
   - You can refer to this [options_Example.json](https://github.com/hsakoh/switchbot-mqtt/blob/main/_docker-compose/options_Example.json)
- Start the container.
- Access port 8099 through your browser.

**Reply by zalmael** [reply]
I forgot to open the ports 🫨 I'm a bit ashamed.
Thank you very much for your help, I am now able to access the app and retrieve my devices. I will have a look later on how it works via my mqtt broker and HA.
For what it is worth and if I may, I'm a bit novice in docker but not in systems and coding, but the guide would benefit to be simplified, a lot of steps are not really mandatory like in my case, pulling directly the image, and for beginners it might be confusing. But don't misunderstand me, I'm really glad and impressed by your work, and I'm really thankful you have tried and managed to help me !!!

**Comment by zalmael** [comment]
Hi, just to let you know that everything is now well set-up and that your add-on is really nice.
I especially like the webhook part which allows to get updates without any increase of the polling rate. Thank you for that and all your help 👍

**Comment by hsakoh** [comment]
Add-on images are published on GitHub Container Registry and DockerHub.

* GitHub Container Registry
  * [switchbotmqtt-amd64](https://github.com/hsakoh/switchbot-mqtt/pkgs/container/switchbotmqtt-amd64)
  * [switchbotmqtt-aarch64](https://github.com/users/hsakoh/packages/container/package/switchbotmqtt-aarch64)
  * [switchbotmqtt-armv7](https://github.com/users/hsakoh/packages/container/package/switchbotmqtt-armv7)

* DockerHub
  * [switchbotmqtt-amd64](https://hub.docker.com/r/hsakoh/switchbotmqtt-amd64)
  * [switchbotmqtt-aarch64](https://hub.docker.com/r/hsakoh/switchbotmqtt-aarch64)
  * [switchbotmqtt-armv7](https://hub.docker.com/r/hsakoh/switchbotmqtt-armv7)

I have updated the installation instructions (added as an advanced installation method).
https://github.com/hsakoh/switchbot-mqtt/wiki/HowToInstall#advanced-add-on-installation-method
